### PR TITLE
Yade: init at 2017.01a

### PR DESCRIPTION
**Pull request by remche**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/physics/yade/default.nix [hunk 1]
```diff
@@ -0,0 +1,121 @@
+with (import <nixpkgs> {});
```
Comment (volth) [inline pkgs/applications/science/physics/yade/default.nix]: please make it a function of the used packages (```{ stdenv, fetchurl, python27Packages etc }:```) like in the .nix files around.
```nixpkgs``` is a bit different than standalone .nix files

File: pkgs/applications/science/physics/yade/default.nix [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+let 
+
+  pythonPackages = python27Packages;
+
+    minieigen = pythonPackages.buildPythonPackage rec {
+      name = "minieigen";
+
+      src = pkgs.fetchurl {
+        url = "https://github.com/eudoxos/minieigen/archive/master.zip";
```
Comment (Mic92) [inline pkgs/applications/science/physics/yade/default.nix]: This is not a stable url. Either refer to a commit or a release. Also use `fetchFromGitHub`.

File: pkgs/applications/science/physics/yade/default.nix [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    };
+
+in 
+
+ {
+
+    minieigen = minieigen;
+
+    yade = stdenv.mkDerivation rec {
```
Comment (Mic92) [inline pkgs/applications/science/physics/yade/default.nix]: this attribute needs to be referenced somewhere. For example in `./pkgs/top-level/all-packages.nix`.

File: pkgs/applications/science/physics/yade/default.nix [hunk 4]
```diff
@@ -27,95 +31,94 @@ let
...
+
+    patches = [ ./cmake.patch ];
+
+    postInstall = ''
+      wrapPythonPrograms
+    '';
+
+    enableParallelBuilding = true;
+
+    system = builtins.currentSystem;
```
Comment (volth) [inline pkgs/applications/science/physics/yade/default.nix]: this seems needless

File: pkgs/applications/science/physics/yade/default.nix [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      platforms   = stdenv.lib.platforms.unix;
+      maintainers = with stdenv.lib.maintainers; [ remche ];
+    };
+
+    nativeBuildInputs = [
+      pkgconfig
+      cmake
+      makeWrapper
+      python2Packages.wrapPython
+      gmp.dev
```
Comment (volth) [inline pkgs/applications/science/physics/yade/default.nix]: does it work without ```.dev```

Comment (remche) [inline pkgs/applications/science/physics/yade/default.nix]: No, had to add the .dev, cmake was not finding it.

File: pkgs/applications/science/physics/yade/default.nix (hunk 4 above)
Comment (remche) [inline pkgs/applications/science/physics/yade/default.nix]: good catch, I'm a total noob… Sorry

File: pkgs/applications/science/physics/yade/default.nix [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      six
+      boost
+      minieigen
+      pillow
+      matplotlib
+    ];
+
+    src = fetchurl
+    {
+      url = "https://launchpad.net/yade/trunk/yade-1.00.0/+download/yade-2017.01a.tar.gz";
```
Comment (volth) [inline pkgs/applications/science/physics/yade/default.nix]: maybe ```fetchbzr``` ?

Comment (remche) [inline pkgs/applications/science/physics/yade/default.nix]: The bzr repository is not used anymore. This is the official link from documentation…

File: pkgs/applications/science/physics/yade/default.nix [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        export LDFLAGS="-L${eigen3_3.out}/lib -l boost_python"
+        export CFLAGS="-I${eigen3_3.out}/include/eigen3"
+      '';
+
+    };
+
+in 
+
+
+  minieigen = minieigen;
```
Comment (qknight) [inline pkgs/applications/science/physics/yade/default.nix]: use `inherit minieigen;` instead

**Comment by Mic92** [comment]
The pull request targets branch `17.03`. We only accept new packages on `master` first, then back-port them maybe.

**Comment by joachifm** [comment]
I expect https://github.com/NixOS/nixpkgs/pull/26873 supercedes this PR, please re-open if I'm mistaken.

**Pull request closed without merging**